x11-misc/clipmenu/clipmenu-6.2.0-r1.ebuild

```diff
@@ -49,7 +49,16 @@ src_install() {
 }
 
 pkg_postinst() {
+	if systemd_is_booted || has_version sys-apps/systemd; then
+		einfo ""
+		einfo "Make sure to import \$DISPLAY when using the systemd unit for clipmenud."
```

Comment:
This is factually only half-true, though?

The issue specifically relates to people whose DE doesn't automatically handle the systemd environment. It's also untrue as a suggested workaround -- /etc/X11/xinit/xinitrc.d/50-systemd-user.sh handles this for you, and it is considered generally correct practice that a custom user session handler should source all the scripts in /etc/X11/xinit/xinitrc.d/

See for example the stock /etc/X11/xinit/xinitrc

```bash
if [ -d /etc/X11/xinit/xinitrc.d ] ; then
 for f in /etc/X11/xinit/xinitrc.d/?* ; do
  [ -x "$f" ] && . "$f"
 done
 unset f
fi
```

Reply:
Thanks for the feedback!

This is upstream's [official recommendation](https://github.com/cdown/clipmenu/tree/8538e01e5c3dff35172789e946d1a60df4ed658b?tab=readme-ov-file#clipmenud), and here I merely copy it. This specific change is not in the released README yet, so I can't directly point to it, so I chose a post-install message. In the planned snapshot ebuild, I may drop this, and point to README.

---

> This is factually only half-true, though?

I see 2 parts of the story/message:

1. Setting expectations with `Make sure to import \$DISPLAY when using the systemd unit for clipmenud.`
2. Offering a sample which could work.

It may be "half-true" in the sense of "this is the issue, fact", and "there are different approaches to solve it, one of them is this".

---

> It's also untrue as a suggested workaround

Hmm, I'm not sure about factually addressing it as "untrue". It does seem to solve the problem for users well enough that upstream accepted a PR to include this in their README.

I accept there may be other viable implementations to achieve the same. I agree your suggested approach of mimicking stock `/etc/X11/xinit/xinitrc` feels more robust. That's also what I would do, if I'd need anything from `/etc/X11/xinit/xinitrc.d`.

---

@eli-schwartz: Your note seems to purely relate to which example to use. In terms of actual change to the ebuild, what do you suggest?

- Use your sample from stock `/etc/X11/xinit/xinitrc` in the post-install message?
- Backport README changes to this clipmenu release, and point to the README?
- Keep the note about expectations, and leave the actual solution to the end-user?
- Anything else?

Perhaps you could suggest this approach upstream as a "more correct" recommendation for their users?

Reply:
I'd like to:
- at least see this somehow highlighted as an issue specific to people not using a DE.
- preferably guide people towards that sample xinitrc rather than the upstream one.

Reply:
> Perhaps you could suggest this approach upstream as a "more correct" recommendation for their users?

Done: https://github.com/cdown/clipmenu/issues/230

Reply:
@eli-schwartz: I updated the post-inst messages, and rebased on top of current master branch. Thanks for the feedback, and please review again.